### PR TITLE
e2e_cloud&common:stop using deprecated framework.ExpectError

### PR DESCRIPTION
**Pull request by wlq1212**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
e2e_cloud:stop using deprecated framework.ExpectError

#### Which issue(s) this PR fixes:

Ref # https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```
NONE
```

### CODE REVIEW

**Review by kerthcet — COMMENTED** [review]
Can you walkthrough the whole **e2e project** for this? We can remove the `framework.ExpectError()` once and for all.

**Comment by wlq1212** [comment]
> Can you walkthrough the whole **e2e project** for this? We can remove the `framework.ExpectError()` once and for all.

sure, will go through and submit  new PR ater. and there are some others are doing this cleanup, will check carefully to avoid duplication

**Comment by wlq1212** [comment]
> Can you walkthrough the whole **e2e project** for this? We can remove the `framework.ExpectError()` once and for all.

sure, will go through and submit  new PR ater. and there are some others are doing this cleanup, will check carefully to avoid duplication

**Comment by kerthcet** [comment]
cc @SataQiu for approval.

File: test/e2e/cloud/gcp/addon_update.go [hunk 1]
```diff
@@ -30,12 +30,13 @@ import (
...
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
+	admissionapi "k8s.io/pod-security-admission/api"
+
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2enetwork "k8s.io/kubernetes/test/e2e/framework/network"
 	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 	e2essh "k8s.io/kubernetes/test/e2e/framework/ssh"
 	imageutils "k8s.io/kubernetes/test/utils/image"
-	admissionapi "k8s.io/pod-security-admission/api"
```
Comment (bart0sh) [inline test/e2e/cloud/gcp/addon_update.go:38]: How moving this import is related to the change?

Comment (wlq1212) [reply test/e2e/cloud/gcp/addon_update.go]: done

File: test/e2e/common/node/privileged.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func (c *PrivilegedPodTestConfig) run(containerName string, expectSuccess bool)
 		framework.ExpectNoError(err,
 			fmt.Sprintf("could not remove dummy1 link: %v", err))
 	} else {
-		framework.ExpectError(err, msg)
```
Comment (SataQiu) [inline test/e2e/common/node/privileged.go:84]: missing something?

**Review by SataQiu — COMMENTED** [review]
/approve

File: test/e2e/common/node/privileged.go (hunk 2 above)
Comment (wlq1212) [reply test/e2e/common/node/privileged.go:84]: done

**Review by SataQiu — COMMENTED** [review]
thanks @wlq1212 
/lgtm

**Pull request merged**